mlir-tensorrt/tensorrt/lib/Target/TensorRTEncodingOpInterface/NetworkEncoder.cpp

```diff
@@ -566,11 +566,12 @@ static void serializeSplatElements(DenseIntOrFPElementsAttr values,
     packed |= ((value & 0x0F) << 4);
     // Fill `data` vector with `packed`
     std::fill_n(reinterpret_cast<uint8_t *>(data.data()), data.size(), packed);
-    return;
+    return llvm::success();
   }
-
-  llvm_unreachable("unsupported data type to convert MLIR splat attribute to "
```

Comment:
This makes no sense. `llvm_reachable` will cause an abort, and program will never proceed to `llvm::errs`. just do 

```
return emitError(UnknownLoc:get(values.getContext()) << "my error message";
```

Reply:
Yes, I replaced `llvm_unreachable` with `llvm::errs` and `llvm::failure`. I will use `emitError` instead.

FYI, several other places could benefit from such a change in `NetworkEncoder.cpp`. I can this as a follow-up PR.

Reply:
Ah, ok, I didn't realize when reviewing that these changes were disconnected from the rest of the MR. We do have an open issue internally for this. Thanks for taking care of it!